File: Algo/Leetcode/024SwapNodesInPairs.cpp
//1. Easy exchange on pairs.
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        if(!head || !head->next) return head;
        ListNode* new_head = new ListNode(-1);
        new_head -> next = head;
        ListNode* move = new_head;
        while(head && head->next){
            move->next = head->next;
            head->next = move->next->next;
            move->next->next = head;
            move = move->next->next;
            head = head->next;
        }
        return new_head->next;
    }
};

File: Algo/Leetcode/024SwapNodesInPairs.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head is None or head.next is None:
            return head

        tmp = head.next.next
        f = ListNode()
        f.next = head.next
        f.next.next = head
        f.next.next.next = self.swapPairs(tmp)

        return f.next